amads/time/meter/__init__.py

```diff
@@ -0,0 +1,619 @@
+"""
+This module serves to map out metrical hierarchies in a number of different ways and
+to express the relationship between
+the notes in
+and
+the hierarchy of
+a metrical cycle.
+
+Uses include identifying notes that traverse metrical levels,
+for analysis (e.g., as a measure of syncopation)
+and notation (e.g., re-notating to reflect the
+within-measure notational conventions).
+"""
+
+from typing import Optional, Union
+
+# ------------------------------------------------------------------------------
+
+
+class MetricalHierarchy:
+    """
+    This class serves to create standard, interoperable representations
+    of metrical hierarchy,
+    centred on a single representation
+    of metrical levels in terms of starts expressed by quarter length
+    from the start of the measure,
+    and (optionally) other representations.
+
+    At least one of the parameters (below) must be specified.
+    These are listed in order of override, e.g., if a full `start_hierarchy` is specified, all else are ignored.
+
+    For a 'quick start', the `time_signature` defaults will serve most use cases, and
+    that along with `levels` should be cover almost all.
+
+    Parameters
+    ----------
+    start_hierarchy
+        Users can specify the `start_hierarchy` directly
```

Comment:
Are we assuming that the highest level of the hierarchy is the measure? Presumably it would be nice to capture hypermetre too?

Reply:
Yes we must and do capture hypermeter. 

When you make a metrical structure with an overall span of 4, that could be a single measure of 4/4 with no hypermeter, but it could also be in 2/4 with a layer of hypermeter.

It is core to this module that we make no such distinction.

Only in the case of building it from a time signature is that specified (by the user).